packages/opentelemetry-instrumentation-google-generativeai/opentelemetry/instrumentation/google_generativeai/utils.py

```diff
@@ -27,3 +29,17 @@ def wrapper(*args, **kwargs):
                 Config.exception_logger(e)
 
     return wrapper
+
+
+def _get_package_name(dist: Distribution) -> Optional[str]:
+    try:
+        return dist.name.lower()
+    except (KeyError, AttributeError):
+        return None
+
+
+installed_packages = {name for dist in distributions() if (name := _get_package_name(dist)) is not None}
+
+
+def is_package_installed(package_name: str) -> bool:
```

Comment:
I think using `importlib.util.find_spec(package_name) is not None` for checking this will be much quicker - just discussed it with @ronensc:
https://github.com/traceloop/openllmetry/pull/2988